lib/processEventDataHelper.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+const moment = require('moment-timezone');
+const _ = require('lodash');
+const Q = require('q');
+
+function checkRequiredFields(cfg, messageBody) {
+  console.log('Checking required fields ...');
+  if (!messageBody.start) {
+    throw new Error('Start Time missing! This field is required!');
+  }
+  if (!messageBody.end) {
+    throw new Error('End Time missing! This field is required!');
+  }
+  if (!cfg.timeZone) {
+    throw new Error('Time Zone missing! This field is required!');
+  }
+  if (!cfg.calendarId) {
+    throw new Error('Calendar ID missing! This field is required!');
+  }
+  if ((cfg.bodyContentType) && (!messageBody.body)) {
+    throw new Error('Body Type provided, but Body Content is missing!');
+  }
+}
+
+
+function formatDate(date, timeZone, format) {
+
+  if (moment(date).isValid()) {
+    let m = moment(date);
+    if (m.creationData().format) {
+      console.log(`Date ${date} has a known ISO-8601 format ${m.creationData().format}.`);
+      return moment.tz(date, timeZone).format(format);
+    }
+    throw new Error(`non ISO-8601 date formats are currently not supported: ${date}.`);
+  }
+
+  if (moment(date, 'x').isValid()) {
```

Comment:
Is it for milliseconds? I think ``moment.js`` can handle them without ``x`` argument. Try it out:

````
moment(new Date().getTime())
````

I'm also missing a test covering dates as milliseconds